pkg/kubectl/cmd/convert.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+)
+
+const (
+	convert_long = `Convert config files between different API versions.
+The command takes filename or stdin as input, and convert it to the format of version
+specified by -v flag. If target version is not specified, convert to latest version.
+Both YAML and JSON formats are accepted.
+`
+	convert_example = `// Convert 'pod.yaml' to latest version and print to stdout.
+$ kubectl convert -i pod.yaml
+
+// Convert 'pod.yaml' to version v1beta3 and output to 'b3pod.yaml'.
+$ kubectl convert -i pod.yaml -v v1beta3 -o b3pod.yaml
+
+// Convert 'pod.yaml' to latest version and rewrite to 'pod.yaml'.
+$ kubectl convert -r pod.yaml
+`
+)
+
+// NewCmdConvert creates a command object for the generic "convert" action, which
+// translates the config file to a given version.
+func NewCmdConvert(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:     "convert -i FILENAME",
+		Short:   "Convert config files between different API versions",
+		Long:    convert_long,
+		Example: convert_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunConvert(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmd.Flags().StringP("input", "i", "-", "Input source; '-' means stdin.")
+	cmd.Flags().StringP("output", "o", "-", "Output destination; '-' means stdout.")
+	cmd.Flags().StringP("rewrite", "r", "", "If nonempty, use this as both input and output.")
+	cmd.Flags().StringP("out-version", "v", latest.Version, "Version to convert input to.")
+	return cmd
+}
+
+// RunConvert implements the generic Convert command
+func RunConvert(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
```

Comment:
OK, I‘ll try to make `convert` in this form. But it will take a little more time:-)